### PR TITLE
Update: Changing on Dependencies

### PR DESCRIPTION
**Pull request by UtkuErdemir**
@aaydin-tr please check this.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -51,10 +51,10 @@
...
   "devDependencies": {
-    "@typescript-eslint/eslint-plugin": "^6.11.0",
-    "@typescript-eslint/parser": "^6.11.0",
-    "bun-types": "latest",
-    "eslint": "^8.53.0",
+    "@typescript-eslint/eslint-plugin": "^6.16.0",
+    "@typescript-eslint/parser": "^6.16.0",
+    "bun-types": "^1.0.20",
+    "eslint": "^8.55.0",
     "eslint-plugin-import": "^2.29.1"
```
Comment (aaydin-tr) [inline package.json:58]: why we are changing dependencies is there any specific reason

File: package.json [hunk 2]
```diff
@@ -63,6 +63,6 @@
   "dependencies": {
     "fast-querystring": "^1.1.2",
     "radix3": "^1.1.0",
-    "reflect-metadata": "^0.1.13"
+    "reflect-metadata": "^0.2.1"
```
Comment (aaydin-tr) [inline package.json:66]: why we are changing dependencies is there any specific reason

File: src/context.ts [hunk 3]
```diff
@@ -280,7 +280,7 @@ export class Context {
     this.res = new Response(null, {
       status: status,
       headers: {
-        ...this.res.headers.toJSON(),
+        ...Object.fromEntries(this.res.headers),
```
Comment (aaydin-tr) [inline src/context.ts:283]: `toJSON()` in bun about 10x faster than `Object.fromEntries(headers.entries())`

File: src/context.ts [hunk 4]
```diff
@@ -321,7 +321,7 @@ export class Context {
     this.res = new Response(jsonData, {
       status: status || this.res.status,
       headers: {
-        ...this.res.headers.toJSON(),
+        ...Object.fromEntries(this.res.headers),
         "Content-Type": "application/json",
```
Comment (aaydin-tr) [inline src/context.ts:325]: same above

File: src/context.ts [hunk 5]
```diff
@@ -371,7 +371,7 @@ export class Context {
   public getResWithoutBody(): Response {
     return new Response(null, {
       status: this.res.status,
-      headers: this.res.headers,
+      headers: Object.fromEntries(this.res.headers),
```
Comment (aaydin-tr) [inline src/context.ts:374]: same above

File: src/context.ts [hunk 6]
```diff
@@ -400,7 +400,7 @@ export class Context {
     this.res = new Response(data, {
       status: status || this.res.status,
       headers: {
-        ...this.res.headers.toJSON(),
+        ...Object.fromEntries(this.res.headers),
```
Comment (aaydin-tr) [inline src/context.ts:403]: same above

File: src/context.ts [hunk 7]
```diff
@@ -426,7 +426,7 @@ export class Context {
...
```
Comment (aaydin-tr) [inline src/context.ts:429]: same above

File: src/context.ts [hunk 8]
```diff
@@ -452,7 +452,7 @@ export class Context {
...
```
Comment (aaydin-tr) [inline src/context.ts:455]: same above

File: package.json (hunk 1 above)
Comment (UtkuErdemir) [reply package.json:58]: Just staying up to date.

File: src/context.ts (hunk 3 above)
Comment (UtkuErdemir) [reply src/context.ts:283]: It may be, but that usage fails pipeline for TS checking

Comment (aaydin-tr) [reply src/context.ts:283]: interesting should't be fail can you rollback `toJSON` and retry

File: package.json (hunk 1 above)
Comment (aaydin-tr) [reply package.json:58]: i guess `latest` can be stay for `bun-types`

File: src/context.ts (hunk 3 above)
Comment (UtkuErdemir) [reply src/context.ts]: I have added back and pipeline is failed again. Could you look that?

Comment (aaydin-tr) [reply src/context.ts]: just rollback all `toJSON` i can force pipeline it should't be fail

**Comment by aaydin-tr** [comment]
`tsc` give error because update i guess best thing to do close this mr and update dependencies one by one what do you say @UtkuErdemir

**Comment by aaydin-tr** [comment]
actually no need to update `devDependencies` right now bun we can update `reflect-metadata`

**Comment by UtkuErdemir** [comment]
@aaydin-tr i have reverted package.json and bunlock but it changed nothing.

**Comment by aaydin-tr** [comment]
closing this mr because no need to change our dependencies right now and its break pipeline

**Pull request closed without merging**